src/extension.ts

```diff
@@ -208,21 +263,44 @@ class RunOnSaveExtension {
     this.showOutputMessage(message);
   }
 
-  /**
-   * Show message in status bar and output channel.
-   * Return a disposable to remove status bar message.
-   */
-  public showStatusMessage(message: string): vscode.Disposable {
-    this.showOutputMessage(message);
-    return vscode.window.setStatusBarMessage(message);
+  public refreshStatus(): void {
+    if (!this._sbStatus) {
+      this._sbStatus = vscode.window.createStatusBarItem(STATUS_BAR_ACTIVITY_ID, vscode.StatusBarAlignment.Left, 100);
+      this._sbStatus.name = 'Run On Save Status';
+      this._sbStatus.command = COMMAND_SHOW_OUTPUT_CHANNEL;
+      this._context.subscriptions.push(this._sbStatus);
+    }
+
+    let unfinishedSyncCount = this._syncCommandQueue.length;
+    if (this._notifyWaitingSyncCommandRunner === undefined && !this._disposed) {
+      // We are not disposed and the runner is not waiting for a command, so it must be running a command.
+      unfinishedSyncCount++;
+    }
+    const asyncCount = this._activeAsyncCommands;
+
+    let state: string;
+
+    const atLeastOneCommandIsRunning = unfinishedSyncCount !== 0 || asyncCount !== 0;
+
+    if (!this.isEnabled()) {
+      // If we are disabled but have unfinished commands, then we are "draining".
```

Comment:
What's the scenario that can result in "draining"? Since commands are killed when extension is disabled, is there any meaningful time that we would see this status?